lib/quiet_quality/version_control_systems/git.rb

```diff
@@ -39,7 +57,48 @@ def self.default_branch(remote:)
       # @return [String] The nearest common ancestor (SHA)
       #
       def comparison_base(sha:, comparison_branch:)
-        git.merge_base(comparison_branch, sha)
+        git.merge_base(comparison_branch, sha).first
+      end
+
+      private def changed_lines_for(diff)
+        GitDiffParser.parse(diff).flat_map do |parsed_diff|
+          parsed_diff.changed_line_numbers.to_set
+        end
+      end
+
+      private def lines_in_file(file_path)
+        File.open(file_path, "r").readline.count
+      end
+
+      private def committed_changes(base, sha)
+        git.diff(base, sha).map do |diff_file|
+          [diff_file.path, changed_lines_for(diff_file.patch)]
+        end.to_h
+      end
+
+      # Combine the changes from the three sources into a single hash, depending on the options given.
+      private def reduced_changes(base:, sha:, include_uncommitted:, include_untracked:)
+        [
+          committed_changes(base, sha),
+          include_uncommitted ? uncommitted_changes(base, sha) : {},
+          include_untracked ? untracked_changes : {}
+        ].reduce do |acc, h|
+          acc.merge(h) { |_key, h1, h2| h1 + h2 }
```

Comment:
Do we need to do some uniquing here? Duplicate entries for line numbers would cause problems elsewhere I think.

Reply:
I don't think we need to - each of the values being merged is a Set of changed line numbers.

Reply:
Oh, didn't notice they were Sets already -.-